File: khalti-android/src/main/java/com/khalti/checkOut/Card/contactForm/ContactFormContract.java
package com.khalti.checkOut.Card.contactForm;

import com.khalti.checkOut.EBanking.helper.BankingData;
import com.khalti.checkOut.helper.Config;
import rx.Observable;

interface ContactFormContract {

    interface View {

        BankingData receiveData();

        void setBankData(String logo, String name, String icon);

        void setButtonText(String text);

        void setEditTextError(String error);

        void setMobile(String mobile);

        void showMessageDialog(String title, String message);

        void showError(String message);

        void showNetworkError();

        void openEBanking(String url);

        void saveConfigInFile(Config config);

        String getPackageName();

        String getContactNumber();

        Observable<Void> setClickListener();

        Observable<CharSequence> setEditTextListener();

        boolean isNetworkAvailable();

        void setPresenter(Presenter presenter);
    }

    interface Presenter {

        void onCreate();

        void onDestroy();

        void onFormSubmitted(boolean isNetwork, String mobile, String bankId, String bankName, Config config);
    }
}

File: khalti-android/src/main/java/com/khalti/checkOut/ebanking/contactForm/ContactFormContract.kt
package com.khalti.checkOut.ebanking.contactForm

import com.khalti.base.LifeCycle
import com.khalti.checkOut.ebanking.helper.BankingData
import com.khalti.checkOut.helper.Config
import com.khalti.signal.Signal
import rx.Observable

interface ContactFormContract {

    interface View {

        val packageName: String

        val contactNumber: String

        val isNetworkAvailable: Boolean

        fun receiveData(): BankingData?

        fun setBankData(logo: String, name: String, icon: String)

        fun setButtonText(text: String)

        fun setEditTextError(error: String?)

        fun setMobile(mobile: String)

        fun showMessageDialog(title: String, message: String)

        fun showError(message: String)

        fun showNetworkError()

        fun openEBanking(url: String)

        fun saveConfigInFile(config: Config)

        fun setOnClickListener(): Map<String, Signal<Any>>

        fun setEditTextListener(): Signal<CharSequence>

        fun setPresenter(presenter: Presenter)
    }

    interface Presenter : LifeCycle {

        fun onFormSubmitted(mobile: String, bankId: String, bankName: String, config: Config)
    }
}
